### PR TITLE
Build script for automatic native dependency management

### PR DESCRIPTION
**Pull request by 0x7CFE**

### CODE REVIEW

File: build.rs [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#[macro_use] extern crate error_chain;
+extern crate curl;
```
Comment (0x7CFE) [inline build.rs]: It's better not to use libraries that have external dependencies. 
Taking this into account, reqwest seems to fit better.

File: build.rs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    let build_mode   = "debug"; // TODO read from env
+
+    let binary_url = format!(
+        "https://github.com/RnDome/bwapi-c/releases/download/v{}/bwapi-c-{}-win32.tar.gz",
+        "0.1.0", "Debug");
+    log_var!(binary_url);
+    let short_file_name = binary_url.split("/").last().unwrap();
+
+    let download_dir = PathBuf::from(&get!("CARGO_MANIFEST_DIR"))
+        .join("target")
```
Comment (kpp) [inline build.rs]: пути надо поправить

File: build.rs [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+        easy.write_function(move |data| {
+            Ok(writer.write(data).unwrap())
+        }).unwrap();
+        easy.perform().unwrap();
+
+        let response_code = easy.response_code().unwrap();
+        match response_code {
+            200 | 302 => { }
+            //302 => { }
```
Comment (kpp) [inline build.rs]: del

File: build.rs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+                panic!("Unexpected response code {} for {}", response_code, url);
+            }
+        }
+    }
+}
+
+fn extract(archive_path: &Path, extract_to: &Path) {
+    let file = File::open(archive_path).unwrap();
+    let unzipped = GzDecoder::new(file).unwrap();
+    let mut archive = Archive::new(unzipped);
```
Comment (kpp) [inline build.rs:46]: TODO: unpack only lib

File: build.rs (hunk 1 above)
Comment (kpp) [reply build.rs]: Have you checked the number of deps in `reqwest` agains `curl`?

Comment (0x7CFE) [reply build.rs]: It's not based on CURL, it's written in Rust completely. It's dependencies are bare Rust too.

Comment (0x7CFE) [reply build.rs]: It was the main reason why I selected reqwest.

File: build.rs [hunk 5]
```diff
@@ -0,0 +1,93 @@
+#[macro_use] extern crate error_chain;
```
Comment (0x7CFE) [inline build.rs]: Since error handling is not used anymore, this dependency is redundant.

File: build.rs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    log_var!(unpacked_dir);
+
+    log!("Obtaining BWAPI-C distribution...");
+    download(&binary_url, &zip_path);
+    println!("cargo:rerun-if-changed={}", zip_path.display());
+
+    log!("Unpacking archive...");
+    extract(&zip_path, &unpacked_dir);
+
+    println!("cargo:rustc-link-search={}", unpacked_dir.display());
```
Comment (0x7CFE) [inline build.rs]: Should point to `unpacked_dir.join("lib")`

**Comment by 0x7CFE** [comment]
Current implementation do not respect cross compilation. Path should be `target/architecture/build_mode/native/`.

**Review by kpp — APPROVED** [review]

**Pull request merged**